### ISSUE TITLE
Improve the build system

### ISSUE THREAD

**Issue by mdymczyk** [h2oai/h2o4gpu#352]
Either clean up the Makefile or redo it as a CMake.

**Comment by pseudotensor** [comment]
How is this going?  Pritvi also wanted the ability to run DAI in windows, and they are only limited by h2o4gpu not working.  cmake should allow windows builds like xgboost.

Also, until cmake build with static cuda, we can't put h2o4gpu into DAI because leads to mixed devel environment (they may have cuda9 and have normal github repo that's at cuda8 and then h2o4gpu can't find libs).

**Comment by mdymczyk** [comment]
Started working on the CMake (heavily based on XGBoost's CMake) but might take me a bit longer since it's my first CMake.

As for bundling the libraries - will do that too but for now, people can either decide on one version or have both CUDA8 and 9 installed, no? It might be a slight inconvenience but not a show stopper?

**Comment by pseudotensor** [comment]
On which cuda, it's that DAI has a default of cuda8 version because of how they build cuda9 as separate branch.  So even people like arno got confused about why things failed with missing libraries and disabled some simple h2o4gpu stuff inside DAI.  Basically, not confusing for docker, but confusing and bad for developers.

**Comment by pseudotensor** [comment]
Please keep all Makefile stuff around -- just add cmake capability for now instead of removing old compile method.  Thanks!

**Comment by pseudotensor** [comment]
Yes, using xgboost as base is very good idea.  h2o4gpu is not too different in overall things needed (python to C++, gpu vs. cpu, nccl/cub and submodules, etc.)

**Comment by vhcg77** [comment]
+1
